app/routes.py

```diff
@@ -1,2 +1,70 @@
-from flask import Blueprint
+from app import db
+from flask import Blueprint, jsonify, abort, make_response, request
+from .models.planet import Planet
 
+bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+def validate_planet(planet_id):
+    try:
+        planet_id = int(planet_id)
+    except:
+        abort(make_response({"message": f"planet {planet_id} invalid"}, 400))
+
+    planet = Planet.query.get(planet_id)
+    if not planet:
+        abort(make_response({"message": f"planet {planet_id} not found"}, 404))
+
+    return planet
+
+def validate_planet_dict(request_body):
+    request_body = dict(request_body)
+    if not (request_body.get("name", False) and request_body.get("description", False)):
+        abort(make_response({'message': 'request body invalid; cannot create planet'}, 400))
+    if request_body.get("rings", 1) == 1:
+        abort(make_response({'message': 'request body invalid; cannot create planet'}, 400))
+
+@bp.route("", methods=["GET"])
+def handle_planets():
+    planets = Planet.query.all()
+    planets_response = []
+    for planet in planets:
+        planets_response.append(planet.to_dict())
+    return jsonify(planets_response)
+
+@bp.route("", methods=["POST"])
+def create_planet():
+    request_body = request.get_json()
+    validate_planet_dict(request_body)
+    new_planet = Planet.from_dict(request_body)
+
+    db.session.add(new_planet)
+    db.session.commit()
+
+    return make_response(f"Planet {new_planet.name} successfully created", 201)
```

Comment:
👀 Make sure to `jsonify` even plain string responses, otherwise flask will return an HTML response rather than JSON.